client/core/trade.go

```diff
@@ -353,6 +351,12 @@ func (t *trackedTrade) negotiate(msgMatches []*msgjson.Match) error {
 		}
 		newFill += qty
 
+		if trade.Filled()+newFill > trade.Quantity {
+			log.Errorf("Match %s would put order %s fill over quantity. Revoking the match.",
+				match.id, t.ID())
+			match.MetaData.Proof.SelfRevoked = true
+		}
+
```

Comment:
Did this happen somehow in testing?
If this does happen though, what's the reasoning for leaving it's quantity in `newFill` if that pushes the trades remaining amount negative? And still adding it `t.matches` is necessary as well?  Would be good to spell out the intent here since it's not obvious.

Reply:
It was just a missing piece of validation that I thought was more important now that we're adding a new path to `negotiate` that uses unsigned `Match`es.

Reply:
`newFill` is only used for logging purposes, but we would be pushing the `Filled` quantity over the order `Quantity`, we just don't act on any of the extra matches.